src/generators/day/index.ts

```diff
@@ -2,15 +2,59 @@ import {Chance} from 'chance';
 
 import createDateGenerator from '../date';
 
+// Avoids pulling in something like moment.js just for this.
+const DAY_REGEX = /^\d{4}-\d{2}-\d{2}$/;
+
+/**
+ * Strictly speaking, the ISO 8601 date format, though often abbreviated as "YYYY-MM-DD", allows for
+ * an "extended" format containing a year with 5 to 6 digits preceded by a +/- sign:
+ * https://tc39.es/ecma262/#sec-expanded-years.
+ *
+ * For convenience, this can be further narrowed to the range that would be expressible as
+ * JavaScript `Dates`: approximately -271821-04-20 to +275760-09-13.
+ *
+ * However, supporting these formats breaks a number of assumptions in Good Eggs code and perhaps in
+ * third-party libraries:
+ * - sorting "date" strings alphanumerically results in a chronological sort
+ * - extracting the "date" part of a `Date` as "YYYY-MM-DD" will always work
+ * - etc.
+ *
+ * It doesn't seem worthwhile to require thousands of calls to `fake.day()` across repos that use
+ * fake-eggs to pass in custom ranges to deal with this, nor to require thousands of calls to
+ * moment.js's `.format('YYYY-MM-DD')` to be updated, since realistically we will never need to
+ * manipulate dates over a thousand years ago nor dates eight thousand years from now, and
+ * realistically our code will not run for eight thousand years. Therefore, we choose to support a
+ * further narrowed range called "Complete date" by the W3C: https://www.w3.org/TR/NOTE-datetime.
+ *
+ * Finally, for convenience (e.g. generating a random day then adding or subtracting days), we clamp
+ * the default range a bit further.
+ */
+export const DEFAULT_FROM = '1001-01-01';
+export const DEFAULT_TO = '9999-01-01';
```

Comment:
I confess that my first reaction to this was that the new chosen boundaries where a bit arbitrary. Like why not `0000-01-01` or `1000-01-01` or `9999-12-31`. But I also confess that it is reasonable to expect that our code would never need to handle dates thousands of years in the future or in the past. So in the end I made peace with it, and now I think this is totally fine.